homeassistant/components/microbees/switch.py

```diff
@@ -0,0 +1,92 @@
+"""Switch integration microBees."""
+import logging
+from typing import Any
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .entity import MicroBeesEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Config entry."""
+    microbees_data = hass.data[DOMAIN][entry.entry_id]
+    microbees = microbees_data.connector
+    bees = microbees_data.bees
+    coordinator = microbees_data.coordinator
+    switches = []
+    for bee in bees:
+        match bee.productID:
+            case 25 | 26 | 27 | 35 | 38 | 46 | 63 | 64 | 65 | 86:
+                for switch in bee.actuators:
+                    switches.append(MBSwitch(switch, bee, microbees, coordinator))
+
+    async_add_entities(switches)
+
+
+class MBSwitch(MicroBeesEntity, SwitchEntity):
+    """Representation of a microBees switch."""
+
+    def __init__(self, act, bee, microbees, coordinator) -> None:
+        """Initialize the microBees switch."""
+        self.act = act
+        self.bee = bee
+        self.microbees = microbees
```

Comment:
This also might be useful to add to the parent class as you would probably use it in more entities